resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -572,6 +573,17 @@ public DnsNameResolverBuilder consolidateCacheSize(int maxNumConsolidation) {
         return this;
     }
 
+    /**
+     * Set the strategy that is used to determine how a {@link DatagramChannel} is used by the resolver.
+     *
+     * @param channelStrategy   the {@link DnsNameResolverChannelStrategy} to use when doing queries.
+     * @return {@code this}
+     */
+    public DnsNameResolverBuilder channelStrategy(DnsNameResolverChannelStrategy channelStrategy) {
```

Comment:
Consider renaming this method and class variable to `datagramChannelStrategy` or `udpChannelStrategy` to make it clear that it applies only to UDP

Reply:
I did name it like this because we also just use `channelFactory(...)` for the `DatagramChannel` stuff. Maybe we should deprecate it and add a new `datagramChannelFactory(...)` tho. WDYT ?

Reply:
See https://github.com/netty/netty/pull/14379